### DISCUSSION TITLE
Axum Integration

### DISCUSSION THREAD

**Discussion by mcmah309** [Totodore/socketioxide#296] [Q&A] [answered]
Is there a reason that the axum example uses a layer rather than a service?
Does both enable connecting over websockets or do they use long polling instead?
I see that 0.12.0 added middleware, is this something that could not be accomplished with towers layers before?

**Comment by Totodore** [comment, marked as answer]
Hey,
- using services or layers doesn't change the connection type. You can use a service on the `/socket.io` path [or any other path if you changed it](https://docs.rs/socketioxide/latest/socketioxide/struct.SocketIoBuilder.html#method.req_path)
- socket.io middlewares are run _in front_ of a namespace. They are different from tower middlewares.
